syscalls.c

```diff
@@ -52,39 +52,49 @@ void *syscalls_mmap(void *ustack)
 {
 	void *vaddr;
 	size_t size;
-	int prot, flags;
-	oid_t *oid;
-	offs_t offs;
+	int prot, flags, fildes;
+	off_t offs;
 	vm_object_t *o;
+	oid_t oid;
 
 	GETFROMSTACK(ustack, void *, vaddr, 0);
 	GETFROMSTACK(ustack, size_t, size, 1);
 	GETFROMSTACK(ustack, int, prot, 2);
 	GETFROMSTACK(ustack, int, flags, 3);
-	GETFROMSTACK(ustack, oid_t *, oid, 4);
-	GETFROMSTACK(ustack, offs_t, offs, 5);
+	GETFROMSTACK(ustack, int, fildes, 4);
+	GETFROMSTACK(ustack, off_t, offs, 5);
 
-	if (oid == OID_NULL) {
-		o = NULL;
-	}
-	else if (oid == OID_PHYSMEM) {
-		o = (void *)-1;
-	}
-	else if (oid == OID_CONTIGUOUS) {
-		o = vm_objectContiguous(size);
-		if (o == NULL) {
-			return (void *)-1;
+	if (fildes == -1) {
```

Comment:
After some thought: perhaps it would be safer to use MAP_ANONYMOUS instead of fd as an indicator?

That is if we decide to always combine MAP_PHYSMEM and MAP_CONTIGUOUS with MAP_ANONYMOUS

Reply:
fd = -1 is not always needed, some external app might not adhere to this rule

Reply:
Ok